addon/components/materialize-switch.js

```diff
@@ -0,0 +1,21 @@
+import Ember from 'ember';
+import layout from '../templates/components/materialize-switch';
+
+export default Ember.Component.extend({
+  layout: layout,
+  classNames: ['switch'],
+  labelOn: 'On',
+  labelOff: 'Off',
+  valueOn: true,
+  valueOff: false,
+  value: false,
+  isChecked: function (key, val) {
```

Comment:
IMO the parsing of `valueOn` vs `valueOff` is the job of the parent component or controller to parse based on a boolean value?

I think it's clearer to keep this just being a proxy of the underlying checkbox and let people react to the change events themselves.